pkg/store/proxy.go

```diff
@@ -344,11 +398,55 @@ func startStreamSeriesSet(
 
 	wg.Add(1)
 	go func() {
+		isFirstRecv := true
 		defer wg.Done()
 		defer close(s.recvCh)
-
 		for {
-			r, err := s.stream.Recv()
+			var cancel context.CancelFunc
+			if s.responseTimeout != 0 {
```

Comment:
Just curious: can we have this timeout for a whole response? Do we really need it per frame? Plus we might allocate a lot here so canceling per stream frame would be nice.

Reply:
If we want to use timeout for a whole response:
1. We must to set such timeout ~query response. in this case we lose opportunity to fast cancel response from slow store (for example)
2. We need to increase channel buffer (for real parallel read from stores)

Reply:
Yes, you are right all is connected and dependent.

I actually think we should increase buffer at some point, to some degree , but not here. Let's keep it per frame.

Reply:
It would be nice to have timeout naming adjusted then to mention frame right now it is:

> If a Store doesn't send any data in this specified duration then a Store will be ignored and partial data will be returned if it's enabled. 0 disables timeout.

Maybe to:

`If a Store doesn't send any frame in this specified duration then a Store will be ignored and partial data will be returned if it's enabled. 0 disables timeout.`

Reply:
I try to increase buffer to fit all response.
Increase buffer affected only usage memory, but not increase RT, independed on slow/fast stores )

Reply:
In my opinion, this memory will be buffered anyway at some point, but anyway, let's not introduce this here. I think we all like this PR in such logic: To make sure we timeout on the first byte from the slow store instead precisely.